core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/PrecomputedChecksummer.java

```diff
@@ -15,15 +15,23 @@
 
 package software.amazon.awssdk.http.auth.aws.internal.signer;
 
+import static software.amazon.awssdk.http.auth.aws.util.SignerConstant.X_AMZ_CONTENT_SHA256;
+
 import java.nio.ByteBuffer;
 import java.util.concurrent.Callable;
 import java.util.concurrent.CompletableFuture;
 import org.reactivestreams.Publisher;
-import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.annotations.SdkInternalApi;
 import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.SdkHttpRequest;
 import software.amazon.awssdk.http.auth.aws.signer.Checksummer;
 
-@SdkProtectedApi
+/**
+ * An implementation of a checksummer that simply passes along a computed value as a checksum. Specifically, this is used in
+ * the cases where the checksum is a pre-defined value that dictates specific behavior by the signer, and flexible checksums is
+ * not enabled for the request (such as aws-chunked payload signing without trailers, unsigned streaming without trailers, etc.).
+ */
+@SdkInternalApi
 public final class PrecomputedChecksummer implements Checksummer {
```

Comment:
Do we need this class? It seems like we can use `FlexibleChecksummer` and pass `ConstantChecksumAlgorithm` right?

Reply:
This is an optimization, using this class means it doesn't need to read the stream like it does with Flexible.

Reply:
Can we rename it to `PrecomputedSha256Checksummer` to make it more clear?